pkg/diagnostics/cluster/rolebindings.go

```diff
@@ -0,0 +1,99 @@
+package cluster
+
+import (
+	"fmt"
+	"io/ioutil"
+
+	kerrs "k8s.io/kubernetes/pkg/api/errors"
+
+	authorizationapi "github.com/openshift/origin/pkg/authorization/api"
+	osclient "github.com/openshift/origin/pkg/client"
+	policycmd "github.com/openshift/origin/pkg/cmd/admin/policy"
+	"github.com/openshift/origin/pkg/diagnostics/types"
+)
+
+// ClusterRoleBindings is a Diagnostic to check that the default cluster role bindings match expectations
+type ClusterRoleBindings struct {
+	ClusterRoleBindingsClient osclient.ClusterRoleBindingsInterface
+	SARClient                 osclient.SubjectAccessReviews
+}
+
+const (
+	ClusterRoleBindingsName = "ClusterRoleBindings"
+)
+
+func (d *ClusterRoleBindings) Name() string {
+	return ClusterRoleBindingsName
+}
+
+func (d *ClusterRoleBindings) Description() string {
+	return "Check that the ClusterRoleBindings are up-to-date"
+}
+
+func (d *ClusterRoleBindings) CanRun() (bool, error) {
+	if d.ClusterRoleBindingsClient == nil {
+		return false, fmt.Errorf("must have client.ClusterRoleBindingsInterface")
+	}
+	if d.SARClient == nil {
+		return false, fmt.Errorf("must have client.SubjectAccessReviews")
+	}
+
+	return userCan(d.SARClient, authorizationapi.AuthorizationAttributes{
+		Verb:     "list",
+		Resource: "clusterrolebindings",
+	})
+}
+
+func (d *ClusterRoleBindings) Check() types.DiagnosticResult {
+	r := types.NewDiagnosticResult(ClusterRoleBindingsName)
+
+	reconcileOptions := &policycmd.ReconcileClusterRoleBindingsOptions{
+		Confirmed:         false,
+		Union:             false,
+		Out:               ioutil.Discard,
+		RoleBindingClient: d.ClusterRoleBindingsClient.ClusterRoleBindings(),
+	}
+
+	changedClusterRoleBindings, err := reconcileOptions.ChangedClusterRoleBindings()
+	if err != nil {
+		r.Error("CRBD1000", err, fmt.Sprintf("Error inspecting ClusterRoleBindings: %v", err))
```

Comment:
> Why not return

This was originally me too.  You should be careful who's code you trust as a template :)